src/workerd/jsg/modules.c++

```diff
@@ -11,9 +11,16 @@ namespace {
 
 class CompileCache {
   // The CompileCache is used to hold cached compilation data for built-in JavaScript modules.
+  //
+  // Importantly, this is a process-lifetime in-memory cache that is only appropriate for
+  // built-in modules.
+  //
+  // The memory-safety of this cache depends on the assumption that entries are never removed
+  // or replaced. If things are ever changed such that entries are removed/replaced, then
+  // we'd likely need to have find return an atomic refcount or something similar.
 public:
   void add(kj::StringPtr key, std::unique_ptr<v8::ScriptCompiler::CachedData> cached) const {
-    cache.lockExclusive()->insert(kj::str(key), kj::mv(cached));
+    cache.lockExclusive()->upsert(kj::str(key), kj::mv(cached), [](auto&) {});
```

Comment:
The upsert callback takes two parameters... does this compile?

Reply:
doh... pushed too fast.